tools/elfdeps.c

```diff
@@ -88,11 +91,106 @@ static const char *mkmarker(GElf_Ehdr *ehdr)
 	    /* alpha doesn't traditionally have 64bit markers */
 	    break;
 	default:
-	    marker = "(64bit)";
+	    classmarker = "(64bit)";
 	    break;
 	}
     }
-    return marker;
+    return classmarker;
+}
+
+static const char *mkarchmarker(GElf_Ehdr *ehdr)
+{
+    char *archmarker = NULL;
+    const char *elf_machine = NULL;
+    const char *elf_endian = NULL;
+    const char *elf_bitsize = NULL;
+
+    /* First the machine type... */
+    switch (ehdr->e_machine) {
+    case EM_SPARC:
```

Comment:
Most(*) 32-bit Linux binaries are normally of type EM_SPARC32PLUS (that's v8+ or v8a), so it should be added.

(*) very unscientific measurement based on http://archive.debian.org/debian-archive/debian/pool/main/g/glibc/libc6-sparcv9_2.3.6.ds1-13etch10_sparc.deb